draft/mson-parameters-headers.md

```diff
@@ -0,0 +1,216 @@
++ Parameters
+    + p
+    + q
+```
+
+Assuming `Parameters` as the name of the *Parameters base type*:
+
+```apib
+# Resource [/resource/{p}{?q}]
++ Parameters (My Parameters)
+
+# Data Structures
+## My Parameters (Parameters)
++ p
++ q
+```
+
+### Headers
+
+The implementation of this RFC should introduce the use of MSON syntax in the
+[Headers section][]. Similarly to Parameters section this Headers section should
+be considered as [MSON type declaration][].
+
+The type being declared inherits from a *Headers base type* unless specified
+otherwise. If another base type is specified it must inherit from the
+*Headers base type*.
+
+The *Headers base type* may the [API Description Namespace][] [`HTTP Headers`][]
+type.
+
+The *Headers base type* should be available as a type for use within any
+[data structures section][].
+
+### Example
+
+```apib
+# Resource [/resource]
+## Retrieve [GET]
++ Response 200
+    + Headers
+        + Server: nginx
+        + `X-RateLimit-Remaining`: 4987 (number) - Remaining rate limit
+```
+
+Assuming `Headers` as the name of the *Headers base type*:
+
+```apib
+# Resource [/resource]
+## Retrieve [GET]
++ Response 200
+    + Headers (My Headers)
+
+# Data Structures
+## My Headers
++ Server: nginx
++ `X-RateLimit-Remaining`: 4987 (number) - The rate limit remaining
+```
+
+## Considerations
+
+### Parameters base type constraints
+
+Using any type in as a member type in a descendant of Parameters base
+type may lead into problems while representing its value in URL. The
+implementation should consider using `application/x-www-form-urlencoded`
+together with [W3C HTML JSON form submission][] in the case where member type
+is of structured type base.
+
+Fore example:
+
+```apib
+# Resource [/resource{?q}]
++ Parameters
+    + q (object)
+        + member1: one
+        + member2: two
+```
+
+Would result into:
+
+```
+/resource?q%5Bmember1%5D=one&p%5Bmember2%5D=two
+```
+
+Additionally, a parser implementation may consider producing a warning in
+situations where member base type may cause an issue.
+
+### Headers base type constraints
+
+An implementation of a *Headers base type* descendant has to consider there may
+be multiple headers of the same name (key) defined. Additionally, if a member
+type of an *Haders base type* descendant is of a structured type a certain form
+and parser warning should be considered similarly to Parameters base type.
+
+### Interoperability with other Data Structures
+
+It should be possible to use any MSON data structure – named type – as a member
+type of Parameters or Headers-based type.
+
+#### Example
+
+```apib
+# My Resource [/resource{?createdAfter}]
++ Attributes
+    + created (Date) - Date when the resource was created
+
+## Retrieve [GET]
++ Parameters
+    + createdAfter (Date, optional) - Retrieve only resources created after this date
+
++ Response 200 (application/json)
+    + Headers
+        + Date (Date) - Date at which the message was originated
```

Comment:
`Date` is good since it's base type is `string`. But what about using a named type whose base type is an object?